core/src/main/java/com/graphhopper/routing/weighting/TurnWeighting.java

```diff
@@ -103,19 +100,19 @@ public long calcMillis(EdgeIteratorState edgeState, boolean reverse, int prevOrN
                 ? calcTurnWeight(origEdgeId, edgeState.getBaseNode(), prevOrNextEdgeId)
                 : calcTurnWeight(prevOrNextEdgeId, edgeState.getBaseNode(), origEdgeId));
 
-        if (turnCostsInSeconds == 0 && origEdgeId == prevOrNextEdgeId)
-            return millis + (long) defaultUTurnCost * 1000;
-
         return millis + 1000 * turnCostsInSeconds;
     }
 
     /**
      * This method calculates the turn weight separately.
-     * Be aware that it always returns 0 turn weight unless a turn weight was explicitly set. This means it does
-     * NOT include default u-turn costs and calling methods have to check for u-turns.
-     * todo: this should be cleaned up in #1520.
      */
     public double calcTurnWeight(int edgeFrom, int nodeVia, int edgeTo) {
+        if (edgeFrom == EdgeIterator.NO_EDGE || edgeTo == EdgeIterator.NO_EDGE) {
```

Comment:
Do you think the EdgeIterator.isValid is more clear here (and in the calcMillis)?

Reply:
yes, changed.